### ISSUE TITLE
Notification and/or a visible flag about POW in queue

### ISSUE THREAD

**Issue by HostFat** [mailchuck/PyBitmessage#189]
Yesterday a friend tried to sent me a message, but he also closed BM just after clicking the send button. So I never received it until today, when I asked him to try again.

Common users are used that after they click the send button everything is done.

On BM is different because the message needs the relative POW to be sent.

So the idea is adding something like an always visible light/flag that indicate that there are one or more POW in queue, or a popup that is showed if the user try to close BM before the queue of POW if completed.

Something like: "There are X work/pow in queue, are you sure that you want to exit from MB? YES/NO"

**Comment by PeterSurda** [comment]
This is a good point. It actually is more complicated. The PoW is uninterruptible, so PyBitmessage will not quit until it finishes. However, the other threads finish in the meantime, so it won't be distributed.

**Comment by krikmo** [comment]
If such feature was to be implemented, I'd like to suggest a simple icon in the bottom-right corner (together with the connections icon) that when hovered (or maybe when clicked) shows the queue of pow and messages to be sent.

And yet, I've grown used to going to the "sent" space after sending messages.

**Comment by PeterSurda** [comment]
This has been addressed in several commits:
- 8e86a7ed6e0fc838400912068eed6e5cd05d5123 - Python and OpenCL PoW is now interruptible (C PoW not yet).
- 638689f7906049cac7b751f4f5cd1b82e1a858fa - The GUI now waits for PoW to finish rather than killing the thread right away. In daemon it doesn't wait (so it should interrupt the PoW, but I haven't tested it)
- 3f089eb19677b74290ca348cf9292acda3c68786 - This is just a fix.

The shutdown procedure in the GUI now not only waits for the PoW to finish, but also it waits slightly longer to give time for the queued messages to be sent out. The way it's now impelemented doesn't guarantee it will be sent, because it would require more complex changes, but it will wait until the connected nodes are notified the object is available and then wait 10 additional seconds, which would normally work out fine.

The progress is now also shown in the status bar so that the user is not wondering why it's frozen. I tried a dialog box with a progress indicator, but it caused actual freezing and maybe it's a bit too obtrusive.

An additional general PoW and distribution indicator somewhere in the status bar may be helpful, I'll look into that later, but the most urgent issue is fixed.
